### PR TITLE
Arc update management

### PR DESCRIPTION
**Pull request by dkirby-ms**
Added update management scenario 

#77

### CODE REVIEW

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+The scenario will show you how to onboard Azure Arc enabled Servers to [Update Management](https://docs.microsoft.com/en-us/azure/automation/update-management/overview), so that you can manage operating system updates for your Azure Arc enabled servers running Windows or Linux.
+
+In this guide, you will create and configure an Azure Automation account and Log Analytics workspace to support Update Management for Azure Arc enabled servers by doing the following:
+
+* Setup a new Log Analytics Workspace and Azure Automation account. 
+
+* Enable Update Management on Azure Arc enabled servers. 
+
+**Note: This guide assumes you already deployed VMs or servers that are running on-premises or other clouds and you have connected them to Azure Arc.**
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:11]: Did you check if we can do this for the Azure VM? we should incl. it the list

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+# Prerequisites
+
+* Clone this repo
+
+    ```terminal
+    git clone https://github.com/microsoft/azure_arc.git
+    ```
+
+* As mentioned, this guide starts at the point where you already deployed and connected VMs or bare-metal servers to Azure Arc. For this scenario, as can be seen in the screenshots below, we will be using an Amazon Web Services (AWS) instance that has been already connected to Azure Arc and is visible as a resource in Azure.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:27]:
```suggestion
* As mentioned, this guide starts at the point where you already deployed and connected VMs or bare-metal servers to Azure Arc. For this scenario, as can be seen in the screenshots below, we will be using an Amazon Web Services (AWS) EC2 instance that has been already connected to Azure Arc and is visible as a resource in Azure.
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+    ![](../img/updateManagement/01.png)
+
+    ![](../img/updateManagement/02.png)
+
+  
+* [Install or update Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Azure CLI should be running version 2.14** or later. Use ```az --version``` to check your current installed version.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:34]:
```suggestion
* [Install or update Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Azure CLI should be running version 2.14 or later. Use ```az --version``` to check your current installed version.
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+* To deploy the ARM template, navigate to the [deployment folder](../updateManagement/) and run the below command:
+
+    ```bash
+    az deployment group create --resource-group <Name of your Azure Resource Group> \
+        --template-file law-template.json \
+        --parameters law-template.parameters.json
+    ```
+
+   ![](../img/updateManagement/04.png)
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:84]: Switch 04/05 png's

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+* When the deployment is complete, you should be able to see your Log Analytics workspace with the Update Management solution from the Azure Portal as shown below.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:86]: add a screenshot showing the resource group as well with the new resource

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+    ![](../img/updateManagement/07.png)
+
+* Select the Log Analytics Agent extension.
+
+    ![](../img/updateManagement/08.png)
+
+* Click Create.
+
+    ![](../img/updateManagement/09.png)
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:108]: align image #

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+* The extension should start to deploy and you should see a screen similar to the example below.
+
+    ![](../img/updateManagement/12.png)
+
+* Click on the Solutions blade of the Log Analytics workspace to check the progress of the Update Management assessment.
+
+    ![](../img/updateManagement/13.png)
+
+* It may take several hours for Update Management to collect enough data to show an assessment for your VM. In the screen below we can see the assessment is being peformed.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:122]: I think that this part needs to be more automated or at least offer an automated deployment as an alternative. See code [here](https://github.com/microsoft/azure_arc/blob/e2c61ff6c84107d3ebc42061dd5a209454364c1d/azure_arc_sqlsrv_jumpstart/azure/arm_template/scripts/Arc_Onboarding.ps1#L266)

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+* From the Automation Account Update Management blade, you should be able to see your scheduled Update deployment from the Deployment Schedules tab. 
+    ![](../img/updateManagement/24.png)
+
+The Update Management solution will now update your Azure Arc enabled servers in the deployment window based on the schedule you defined. There is a lot more you can do with [Update Management](https://docs.microsoft.com/en-us/azure/automation/update-management/overview) that is outside the scope of this scenario. Review the [documentation](https://docs.microsoft.com/en-us/azure/automation/update-management/overview) for more information. 
+
+# Clean up environment
+
+Complete the following steps to clean up your environment.
+
+* Remove the virtual machines from each environment by following the teardown instructions from each guide.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:169]: See my comment on the Azure VM

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+
+    ```bash
+    az monitor log-analytics workspace delete --resource-group <Name of the Azure Resource Group> --workspace-name <Log Analytics Workspace Name> --yes
+    ```
+
+* Delete the Automation Account from the Azure portal.
+
+    ![](../img/updateManagement/25.png)
+
+* Delete the resource group.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:186]: add a single screenshot for removing the Resource Group

**Review by likamrat — CHANGES_REQUESTED** [review]
- Automating Log Analytics Agent deployment
- Screenshots order alignment
- Screenshots additions

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md (hunk 4 above)
Comment (dkirby-ms) [reply azure_arc_servers_jumpstart/docs/arc_updateManagement.md:84]: these are just numbered out of order, the screens themselves are in the right order. If you insist, I'll rename all the screenshots so the numbers line up but as they are now the screens themselves are in the right order.

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md (hunk 7 above)
Comment (dkirby-ms) [reply azure_arc_servers_jumpstart/docs/arc_updateManagement.md:122]: In the example we are starting with an existing Arc enabled VM so deploying the agent is most straightforward for the user by going through the portal. I will point out references to other options to consider for other scenarios.

Comment (dkirby-ms) [reply azure_arc_servers_jumpstart/docs/arc_updateManagement.md]: updated, all one arm template now. opted not to give two approaches and instead its just done through a single approach using a single arm template with nested deployments.

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+**Note: This guide assumes you already deployed VMs or servers that are running on-premises or other clouds and you have connected them to Azure Arc.**
+
+**If you haven't, this repository offers you a way to do so in an automated fashion:**
+- **[GCP Ubuntu VM](gcp_terraform_ubuntu.md) / [GCP Windows VM](gcp_terraform_windows.md)**
+- **[AWS Ubuntu VM](aws_terraform_ubuntu.md)**
+- **[Azure Ubuntu VM](azure_arm_template_linux.md) / [Azure Windows VM](azure_arm_template_win.md)**
+- **[VMware Ubuntu VM](vmware_terraform_ubuntu.md) / [VMware Windows Server VM](vmware_terraform_winsrv.md)**
+- **[Local Ubuntu VM](local_vagrant_ubuntu.md) / [Local Windows VM](local_vagrant_windows.md)**
+
+# Prerequisites
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:20]:
```suggestion
## Prerequisites
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+        --parameters law-template.parameters.json
+    ```
+
+   ![](../img/updateManagement/05.png)
+
+* When the deployment is complete, you should be able to see the resource group with your Log Analytics workspace, automation account and Update Management solution from the Azure Portal. Drilling into the Log Analytics workspace Solutions blade should show the Update Management solution.
+
+    ![](../img/updateManagement/06.png)
+
+# Confirm that the Update Management solution is deployed on your Azure Arc enabled server.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:93]:
```suggestion
## Confirm that the Update Management solution is deployed on your Azure Arc enabled server.
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+
+* When the assessment is complete, you will see an option to "View summary" on the Update Management blade.
+
+    ![](../img/updateManagement/15.png)
+
+* Click View Summary and then click again to drill into the Update Management assessment. In the below example we can see there are updates missing on our Azure Arc enabled server.
+
+    ![](../img/updateManagement/16.png)
+
+# Schedule an Update
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:111]:
```suggestion
## Schedule an Update
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+* Finally, provide a name for your Update deployment and then click Create.
+
+    ![](../img/updateManagement/23.png)
+
+* From the Automation Account Update Management blade, you should be able to see your scheduled Update deployment from the Deployment Schedules tab. 
+    ![](../img/updateManagement/24.png)
+
+The Update Management solution will now update your Azure Arc enabled servers in the deployment window based on the schedule you defined. There is a lot more you can do with [Update Management](https://docs.microsoft.com/en-us/azure/automation/update-management/overview) that is outside the scope of this scenario. Review the [documentation](https://docs.microsoft.com/en-us/azure/automation/update-management/overview) for more information. 
+
+# Clean up environment
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:142]:
```suggestion
## Clean up environment
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+
+    - *[GCP Ubuntu VM](gcp_terraform_ubuntu.md) / [GCP Windows VM](gcp_terraform_windows.md)*
+    - *[AWS Ubuntu VM](aws_terraform_ubuntu.md)*
+    - *[Azure Ubuntu VM](azure_arm_template_linux.md) / [Azure Windows VM](azure_arm_template_win.md)*
+    - *[VMware Ubuntu VM](vmware_terraform_ubuntu.md) / [VMware Windows Server VM](vmware_terraform_winsrv.md)*
+    - *[Local Ubuntu VM](local_vagrant_ubuntu.md) / [Local Windows VM](local_vagrant_windows.md)*
+
+* Remove the Log Analytics workspace by executing the following script in AZ CLI. Provide the workspace name you used when creating the Log Analytics Workspace.
+
+    ```bash
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:156]:
```suggestion
    ```terminal
```

File: azure_arc_servers_jumpstart/docs/arc_updateManagement.md [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+    az monitor log-analytics workspace delete --resource-group <Name of the Azure Resource Group> --workspace-name <Log Analytics Workspace Name> --yes
+    ```
+
+* Delete the Automation Account from the Azure portal.
+
+    ![](../img/updateManagement/25.png)
+
+* Delete the resource group.
+
+    ```bash
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/arc_updateManagement.md:166]:
```suggestion
    ```terminal
```

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**